ukvm/ukvm_elf.c

```diff
@@ -197,6 +217,17 @@ void ukvm_elf_load(const char *file, uint8_t *mem, size_t mem_size,
             if (mprotect(daddr, _end - paddr, PROT_EXEC | PROT_READ) == -1)
                 goto out_error;
         }
+
+#ifdef __aarch64__
+        ukvm_region_list->regions[ph_i].slot = ph_i;
+        ukvm_region_list->regions[ph_i].guest_phys_addr = (uint64_t)paddr;
+        ukvm_region_list->regions[ph_i].memory_size = _end - paddr;
+        ukvm_region_list->regions[ph_i].userspace_addr = (uint64_t)daddr;
+        ukvm_region_list->regions[ph_i].flags = (phdr[ph_i].p_flags & PF_X)
+                                                ? KVM_MEM_READONLY
+                                                : 0;
+        ukvm_region_list->count++;
+#endif
```

Comment:
The ELF loader cannot depend on hypervisor specifics. If I understand the `ukvm_region_list` correctly, it serves two purposes for ARM64:

1. To have a way of creating the MMIO region for hypercalls.
2. To set up the guest page tables to match the ELF program headers, ie. in practice to get a read-only mapping for the first program header (containing `.text`, `.rodata`, `.eh_frame`).

Thinking about simplifying this:

1. Is always at a fixed address, so can be done by an appropriate `#ifdef` in `ukvm_hv_init()`.
2. For x86_64 we cheat: The guest page tables are all RWX and set up in a simple loop. However, we still get a read-only mapping since that's done by the `mprotect()` call in the ELF loader. The only difference is that instead of a guest-side fault on attempting to write to those pages we get an `EFAULT` back from `ioctl(KVM_RUN)` instead.

Could we use the same approach for ARM64, or is there some other need for `ukvm_region_list` that I've missed?